src/FieldWarning/Assets/Units/Component/Weapon/Cannon.cs

```diff
@@ -57,21 +58,23 @@ private void FireWeapon(TargetTuple target)
                 int roll = rnd.Next(1, 100);
 
                 // HIT
-                if (roll < _data.Accuracy) {
-                    target.Enemy.HandleHit(_data.Damage);
+                if (roll <= _data.Accuracy) {
+                    Debug.LogWarning("Cannon shell dispersion is not implemented yet");
+                    target.Enemy.HandleHit(_data.Damage, _distance, null);
                 }
             } else {
                 // TODO: fire pos damage not implemented
             }
         }
 
-        public bool TryShoot(TargetTuple target, float deltaTime)
+        public bool TryShoot(TargetTuple target, float deltaTime, float distanceToTarget)
         {
             _reloadTimeLeft -= deltaTime;
             if (_reloadTimeLeft > 0)
                 return false;
 
             _reloadTimeLeft = _data.ReloadTime;
+            _distance = distanceToTarget;
```

Comment:
Pass this as an argument to FireWeapon or wherever you need it, do not make it a member variable. State is bad and we want to minimize it when possible.